packages/babel-generator/src/printer.ts

```diff
@@ -323,6 +323,13 @@ class Printer {
     this._buf.withSource(prop, loc, cb);
   }
 
+  withIdentifierName(identifierName: string, cb: () => void): void {
+    const pos = this._buf._sourcePosition;
+    pos.identifierName = identifierName;
+    cb();
```

Comment:
These closure allocations are probably more expensive that the property access, no? Could we just queue the name, and have it unset after making the next mark?